uktrade/data-hub-frontend
# Accessibility work: Format form inputs to be accessible

## Description of change

As part of the accessibility work, some of the form inputs needed addressing. In general, the convention we have decided to follow (as there were many conflicting articles of advice) are:
- All form inputs should have matching `for` and `id` attributes for their labels and input pairs. 
- If a form input can't have `for` and `id` attributes, then the input itself should have an `aria-label` attribute. 
- All checkboxes and radio inputs must be encased in a `Fieldset` and have a `legend`. 

The way the `FieldWrapper` component works, is that It applies the `htmlFor` and the input within it has the matching `id` applied to it. If you supply a `legend`, it'll wrap the input in a `Fieldset` - this means in order to have a` Fieldset`, you have to supply a legend. 

Some pages have different requirements when it comes to general labelling. For example, the add a new company page's first step has just one radio question with a bigger question. Whereas, another page that uses the radios within a form has a smaller question. This raised a need for a customisable legend within the `Fieldset` to cater for these two types of design. He's called bigLegend but we can rename. 

This PR fixes a couple of these problems. 
- The FieldWrapper component takes a boolean prop called `bigLegend` which will change the style applied to the legend. 
- The FieldCheckboxes and FieldRadios components have been amended to pass down this bigLegend prop. 
- The three above components have also had `aria-labels` applied to their inputs. 
- The sample pages that use these react components have been amended so that they use the legend / bigLegend props. 
- The pages that don't use the 'Field' components have been amended and modified to follow the conventions above. 

## Test instructions

Visually, you should see nothing change, everything should look the same.  The pages that have been modified in this PR should have inputs that either have `for` and `id` for the labels and the inputs, and `fieldset` and `legends` for Checkboxes and Radios. You should be able to see these in the dev console when inspecting an element. 

The pages that have been modified are: 

- http://localhost:3000/companies/create
- http://localhost:3000/interactions/6724a277-d767-429a-a19a-680f270b32d9/edit
- http://localhost:3000/ ( the filters on the my companies lists tab, the pipelines tab, the referrals tab )

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
